packs/vsphere/actions/vmwarelib/actions.py

```diff
@@ -24,14 +24,36 @@
 class BaseAction(Action):
     def __init__(self, config):
         super(BaseAction, self).__init__(config)
-        self.si = self._connect()
+        if config is None:
+            raise ValueError("No connection configuration details found")
+        if "vsphere" in config:
+            if config['vsphere'] is None:
+                raise ValueError("'vsphere' config defined but empty.")
+            else:
+                pass
+        elif set(['host', 'port', 'user', 'passwd']).issubset(config):
+            pass
+        else:
+            raise ValueError("Check Connection configuration details")
+
+    def establish_connection(self, vsphere):
+        self.si = self._connect(vsphere)
         self.si_content = self.si.RetrieveContent()
 
-    def _connect(self):
-        si = connect.SmartConnect(host=self.config['host'],
-                                  port=self.config['port'],
-                                  user=self.config['user'],
-                                  pwd=self.config['passwd'])
+    def _connect(self, vsphere):
+        if vsphere:
+            connection = self.config['vsphere'].get(vsphere)
```

Comment:
Addd better validation around connection and incomplete configuration details. passed the the exception message up the tree to shw on the output.